### ISSUE TITLE
无法使用vllm 部署qwen3-vl-embedding

### ISSUE THREAD

**Issue by wyk777777** [xorbitsai/inference#4644]
### System Info / 系統信息

Package                           Version
--------------------------------- ------------------
absl-py                           2.3.1
accelerate                        1.10.0
addict                            2.4.0
aiofiles                          24.1.0
aiohappyeyeballs                  2.6.1
aiohttp                           3.13.3
aioprometheus                     23.12.0
aiosignal                         1.4.0
aliyun-python-sdk-core            2.16.0
aliyun-python-sdk-kms             2.16.5
annotated-doc                     0.0.4
annotated-types                   0.7.0
anthropic                         0.71.0
antlr4-python3-runtime            4.9.3
anyascii                          0.3.3
anyio                             4.10.0
apache-tvm-ffi                    0.1.8.post2
argbind                           0.3.9
argcomplete                       3.6.2
astor                             0.8.1
async-timeout                     5.0.1
attrdict                          2.0.1
attrs                             25.3.0
audioread                         3.0.1
av                                15.1.0
babel                             2.17.0
bcrypt                            4.3.0
beautifulsoup4                    4.14.2
bitsandbytes                      0.48.1
black                             25.9.0
blake3                            1.0.7
blis                              1.3.0
blobfile                          3.1.0
boto3                             1.40.50
botocore                          1.40.50
Brotli                            1.1.0
cached_path                       1.8.0
cachetools                        6.2.0
catalogue                         2.0.10
cbor                              1.0.0
cbor2                             5.7.0
cdifflib                          1.2.9
certifi                           2025.8.3
cffi                              1.17.1
charset-normalizer                3.4.3
chattts                           0.2.4
click                             8.1.8
cloudpathlib                      0.23.0
cloudpickle                       3.1.1
cn2an                             0.5.23
coloredlogs                       15.0.1
compressed-tensors                0.13.0
confection                        0.1.5
conformer                         0.3.2
contourpy                         1.3.3
contractions                      0.1.73
controlnet_aux                    0.0.10
crcmod                            1.7
cryptography                      45.0.6
csvw                              3.7.0
cuda-bindings                     13.1.1
cuda-pathfinder                   1.3.3
cuda-python                       13.1.1
cupy-cuda12x                      13.6.0
curated-tokenizers                0.0.9
curated-transformers              0.1.1
cycler                            0.12.1
cymem                             2.0.11
Cython                            3.1.4
datamodel-code-generator          0.35.0
datasets                          4.2.0
dateparser                        1.1.8
decorator                         5.2.1
DeepCache                         0.1.1
Deprecated                        1.2.18
depyf                             0.20.0
diffusers                         0.35.1
dill                              0.4.0
diskcache                         5.6.3
Distance                          0.1.3
distro                            1.9.0
dlinfo                            2.0.0
dnspython                         2.8.0
docopt                            0.6.2
docstring_parser                  0.17.0
ecdsa                             0.19.1
editdistance                      0.8.1
einops                            0.8.1
einx                              0.3.0
email-validator                   2.3.0
encodec                           0.1.1
espeakng-loader                   0.2.4
eva-decord                        0.6.1
fastapi                           0.128.6
fastapi-cli                       0.0.13
fastapi-cloud-cli                 0.3.1
fastrlock                         0.8.3
ffmpy                             0.6.1
filelock                          3.19.1
fire                              0.7.1
FlagEmbedding                     1.3.5
flashinfer-python                 0.6.1
flatbuffers                       25.9.23
flatten-dict                      0.4.2
fonttools                         4.60.1
frozendict                        2.4.6
frozenlist                        1.7.0
fsspec                            2025.7.0
fugashi                           1.5.1
funasr                            1.2.7
future                            1.0.0
fvcore                            0.1.5.post20221221
g2p-en                            2.1.0
gdown                             5.2.0
genson                            1.3.0
gguf                              0.17.1
google-api-core                   2.26.0
google-auth                       2.41.1
google-cloud-core                 2.4.3
google-cloud-storage              2.19.0
google-crc32c                     1.7.1
google-resumable-media            2.7.2
googleapis-common-protos          1.70.0
gradio                            5.42.0
gradio_client                     1.11.1
groovy                            0.1.2
grpcio                            1.78.0
grpcio-reflection                 1.78.0
gruut                             2.4.0
gruut-ipa                         0.13.0
gruut_lang_de                     2.0.1
gruut_lang_en                     2.0.1
gruut_lang_es                     2.0.1
gruut_lang_fr                     2.0.2
h11                               0.16.0
hf-xet                            1.1.7
httpcore                          1.0.9
httptools                         0.7.1
httpx                             0.28.1
httpx-sse                         0.4.3
huggingface-hub                   0.34.4
humanfriendly                     10.0
hydra-core                        1.3.2
HyperPyYAML                       1.2.2
idna                              3.10
ijson                             3.4.0.post0
imageio                           2.37.0
imageio-ffmpeg                    0.6.0
importlib_metadata                8.7.0
importlib_resources               6.5.2
inflect                           7.5.0
inscriptis                        2.6.0
interegular                       0.3.3
iopath                            0.1.10
ir_datasets                       0.5.11
isodate                           0.7.2
isort                             6.1.0
jaconv                            0.4.0
jamo                              0.4.1
jieba                             0.42.1
Jinja2                            3.1.6
jiter                             0.10.0
jj-pytorchvideo                   0.1.5
jmespath                          0.10.0
joblib                            1.5.2
json5                             0.12.1
jsonlines                         1.2.0
jsonschema                        4.25.1
jsonschema-specifications         2025.9.1
julius                            0.2.7
kaldifst                          1.7.17
kaldiio                           2.18.1
kiwisolver                        1.4.9
kokoro                            0.9.4
langcodes                         3.5.0
langdetect                        1.0.9
language_data                     1.3.0
language-tags                     1.2.0
lark                              1.2.2
lazy_loader                       0.4
librosa                           0.11.0
lightning                         2.5.5
lightning-utilities               0.15.2
llguidance                        1.3.0
llvmlite                          0.44.0
lm-format-enforcer                0.11.3
loguru                            0.7.3
loralib                           0.1.2
lxml                              6.0.2
lz4                               4.4.4
marisa-trie                       1.3.1
Markdown                          3.9
markdown-it-py                    4.0.0
MarkupSafe                        3.0.2
matplotlib                        3.10.7
mcp                               1.26.0
mdurl                             0.1.2
mecab-python3                     1.0.10
misaki                            0.9.4
mistral_common                    1.9.0
model-hosting-container-standards 0.1.13
modelscope                        1.29.0
more-itertools                    10.8.0
mpmath                            1.3.0
msgpack                           1.1.2
msgspec                           0.19.0
multidict                         6.6.4
multiprocess                      0.70.16
munch                             4.0.0
murmurhash                        1.0.13
mypy_extensions                   1.1.0
natsort                           8.4.0
nemo_text_processing              1.0.2
networkx                          3.5
ninja                             1.13.0
nltk                              3.9.2
num2words                         0.5.14
numba                             0.61.2
numpy                             2.2.6
nvidia-cublas-cu12                12.8.4.1
nvidia-cuda-cupti-cu12            12.8.90
nvidia-cuda-nvrtc-cu12            12.8.93
nvidia-cuda-runtime-cu12          12.8.90
nvidia-cudnn-cu12                 9.10.2.21
nvidia-cudnn-frontend             1.18.0
nvidia-cufft-cu12                 11.3.3.83
nvidia-cufile-cu12                1.13.1.3
nvidia-curand-cu12                10.3.9.90
nvidia-cusolver-cu12              11.7.3.90
nvidia-cusparse-cu12              12.5.8.93
nvidia-cusparselt-cu12            0.7.1
nvidia-cutlass-dsl                4.3.5
nvidia-ml-py                      12.575.51
nvidia-nccl-cu12                  2.27.5
nvidia-nvjitlink-cu12             12.8.93
nvidia-nvshmem-cu12               3.3.20
nvidia-nvtx-cu12                  12.8.90
omegaconf                         2.3.0
onnxruntime                       1.23.1
openai                            1.99.9
openai-harmony                    0.0.4
opencv-python-headless            4.13.0.92
optimum                           2.0.0
ordered-set                       4.1.0
orjson                            3.11.2
ormsgpack                         1.11.0
oss2                              2.19.1
outlines_core                     0.2.11
packaging                         25.0
pandas                            2.3.1
parameterized                     0.9.0
partial-json-parser               0.2.1.1.post6
passlib                           1.7.4
pathspec                          0.12.1
peft                              0.17.0
phonemizer-fork                   3.3.2
pillow                            11.3.0
pip                               25.3
platformdirs                      4.5.0
pooch                             1.8.2
portalocker                       3.2.0
preshed                           3.0.10
proces                            0.1.7
prometheus_client                 0.23.1
prometheus-fastapi-instrumentator 7.1.0
propcache                         0.3.2
proto-plus                        1.26.1
protobuf                          6.33.5
psutil                            7.0.0
py-cpuinfo                        9.0.0
pyahocorasick                     2.3.0
pyarrow                           21.0.0
pyasn1                            0.6.1
pyasn1_modules                    0.4.2
pybase16384                       0.3.8
pybase64                          1.4.2
pycountry                         24.6.1
pycparser                         2.22
pycryptodome                      3.23.0
pycryptodomex                     3.23.0
pydantic                          2.12.5
pydantic_core                     2.41.5
pydantic-extra-types              2.10.6
pydantic-settings                 2.12.0
pydub                             0.25.1
Pygments                          2.19.2
PyJWT                             2.11.0
pykakasi                          2.3.0
pyloudnorm                        0.1.1
pynini                            2.1.5
pynndescent                       0.5.13
pynvml                            12.0.0
pyparsing                         3.2.5
pypinyin                          0.55.0
pypinyin-dict                     0.9.0
PySocks                           1.7.1
python-crfsuite                   0.9.11
python-dateutil                   2.9.0.post0
python-dotenv                     1.1.1
python-jose                       3.5.0
python-json-logger                4.0.0
python-multipart                  0.0.20
pytokens                          0.1.10
pytorch-lightning                 2.5.5
pytorch-wpe                       0.0.1
pytz                              2025.2
pyworld                           0.3.5
PyYAML                            6.0.2
pyzmq                             27.1.0
quantile-python                   1.1
qwen-omni-utils                   0.0.8
qwen-vl-utils                     0.0.14
randomname                        0.2.1
ray                               2.50.0
rdflib                            7.2.1
referencing                       0.36.2
regex                             2025.7.34
requests                          2.32.4
rfc3986                           1.5.0
rich                              13.9.4
rich-toolkit                      0.15.1
rignore                           0.7.0
rpds-py                           0.27.1
rsa                               4.9.1
ruamel.yaml                       0.18.15
ruamel.yaml.clib                  0.2.14
ruff                              0.12.9
s3transfer                        0.14.0
sacremoses                        0.1.1
safehttpx                         0.1.6
safetensors                       0.6.2
scikit-image                      0.25.2
scikit-learn                      1.7.2
scipy                             1.16.1
segments                          2.3.0
semantic-version                  2.10.0
sentence-transformers             5.1.1
sentencepiece                     0.2.1
sentry-sdk                        2.41.0
setproctitle                      1.3.6
setuptools                        78.1.1
shellingham                       1.5.4
silero-vad                        6.0.0
six                               1.17.0
smart_open                        7.3.1
sniffio                           1.3.1
socksio                           1.0.0
soundfile                         0.13.1
soupsieve                         2.8
soxr                              1.0.0
spacy                             3.8.7
spacy-curated-transformers        0.3.1
spacy-legacy                      3.0.12
spacy-loggers                     1.0.5
srsly                             2.5.1
sse-starlette                     3.0.2
starlette                         0.52.1
supervisor                        4.3.0
sympy                             1.14.0
tabulate                          0.9.0
tblib                             3.1.0
tensorboard                       2.20.0
tensorboard-data-server           0.7.2
tensorboardX                      2.6.4
termcolor                         3.1.0
textsearch                        0.0.24
thinc                             8.3.6
threadpoolctl                     3.6.0
tifffile                          2025.10.4
tiktoken                          0.12.0
timm                              1.0.19
tokenizers                        0.22.1
tomli                             2.3.0
tomlkit                           0.13.3
torch                             2.9.1
torch-complex                     0.4.4
torchaudio                        2.9.1
torchdiffeq                       0.2.5
torchmetrics                      1.8.2
torchvision                       0.24.1
tqdm                              4.67.1
transformers                      4.57.1
transformers-stream-generator     0.0.5
trec-car-tools                    2.6
triton                            3.5.1
typeguard                         4.4.4
typer                             0.16.0
typing_extensions                 4.14.1
typing-inspection                 0.4.2
tzdata                            2025.2
tzlocal                           5.3.1
umap-learn                        0.5.9.post2
unidic-lite                       1.0.8
unlzw3                            0.2.3
uritemplate                       4.2.0
urllib3                           2.5.0
uv                                0.9.2
uvicorn                           0.35.0
uvloop                            0.21.0
vector-quantize-pytorch           1.17.3
verovio                           5.6.0
vllm                              0.15.0
vocos                             0.1.0
warc3-wet                         0.2.5
warc3-wet-clueweb09               0.2.5
wasabi                            1.1.3
watchfiles                        1.1.0
weasel                            0.4.1
websockets                        15.0.1
Werkzeug                          3.1.3
wetext                            0.1.2
WeTextProcessing                  1.0.3
wget                              3.2
wheel                             0.45.1
wrapt                             1.17.3
x-transformers                    2.8.4
xformers                          0.0.32.post1
xgrammar                          0.1.29
xinference                        2.2.0
xllamacpp                         0.2.2
xoscar                            0.9.3
xxhash                            3.6.0
yacs                              0.1.8
yarl                              1.20.1
zipp                              3.23.0
zlib-state                        0.1.10

### Running Xinference with Docker? / 是否使用 Docker 运行 Xinfernece？

- [ ] docker / docker
- [x] pip install / 通过 pip install 安装
- [ ] installation from source / 从源码安装

### Version info / 版本信息

xinference 2.2.0

### The command used to start Xinference / 用以启动 xinference 的命令

nohup xinference-local \
  --host 0.0.0.0 \
  --port 9997 \
  --auth-config auth.json \
  > /xinference.log 2>&1 &

### Reproduction / 复现过程

启动服务后访问webui（vllm版本为0.15.0）

<img width="2251" height="1077" alt="Image" src="https://github.com/user-attachments/assets/e7f9fea6-cb9d-4eb0-8194-9427d9a3f9f7" />

### Expected behavior / 期待表现

可以vllm启动

**Comment by ace-xc** [comment]
hello，尝试重启下环境或者是因为虚拟环境的问题，如果原有的环境中已经存在对应的依赖，先尝试原有的环境，将Model Virtual Environments暂时设置为False。我这边复现了你遇到的情况，使用的是 vllm==0.15.0，并且已经定位到具体问题。我已经将对应的修改稍后提交至 PR。或者你也可以先在本地环境中把相关代码注释掉/使用可修复的代码，作为临时解决方案。
方案一：注释
```
#  if model_family.model_name.startswith("Qwen3-VL-Embedding"):
#            return False, "Qwen3-VL embedding requires vLLM>=0.14.0"
```

方案二：将原有代码替换
```
# 修改前
if model_family.model_name.startswith("Qwen3-VL-Embedding"):
    return False, "Qwen3-VL embedding requires vLLM>=0.14.0"

# 修改后
if model_family.model_name.startswith("Qwen3-VL-Embedding"):
    from packaging import version
    import vllm
    if version.parse(vllm.__version__) < version.parse("0.14.0"):
        return False, f"Qwen3-VL embedding requires vLLM>=0.14.0, current: {vllm.__version__}"
```

<img width="1407" height="731" alt="Image" src="https://github.com/user-attachments/assets/d6b32509-aa65-4555-9a8c-30ccd72b3f4a" />

<img width="2880" height="1530" alt="Image" src="https://github.com/user-attachments/assets/e515aa90-942e-4151-bc6b-d233ab4f194d" />

<img width="1403" height="1318" alt="Image" src="https://github.com/user-attachments/assets/1f9d1b67-bbde-4bb9-b6bf-31a6e04eb106" />

<img width="2880" height="1530" alt="Image" src="https://github.com/user-attachments/assets/a9ecb19b-5f76-4a01-9e29-5e30c7e4276c" />
